lambdas/data-migration2/src/index.ts

```diff
@@ -1,52 +1,70 @@
 import { getKnexClient } from '@cumulus/db';
-import { MigrationSummary } from '@cumulus/types/migration';
 import Logger from '@cumulus/logger';
+import {
+  DataMigration2Summary,
+  DataMigration2HandlerEvent,
+  MigrationSummary,
+} from '@cumulus/types/migration';
 
 import { migrateExecutions } from './executions';
 import { migrateGranulesAndFiles } from './granulesAndFiles';
 import { migratePdrs } from './pdrs';
 
 const logger = new Logger({ sender: '@cumulus/data-migration2' });
-export interface HandlerEvent {
-  env?: NodeJS.ProcessEnv
-}
 
-export const handler = async (event: HandlerEvent): Promise<MigrationSummary> => {
+export const handler = async (
+  event: DataMigration2HandlerEvent
+): Promise<MigrationSummary> => {
   const env = event.env ?? process.env;
+  const migrationsToRun = event.migrationsList ?? ['executions', 'granules', 'pdrs'];
+
   const knex = await getKnexClient({ env });
 
   try {
-    const executionsMigrationSummary = await migrateExecutions(env, knex);
-    const granulesAndFilesMigrationSummary = await migrateGranulesAndFiles(env, knex);
-    const pdrsMigrationSummary = await migratePdrs(env, knex);
+    const migrationSummary: DataMigration2Summary = {};
+
+    if (migrationsToRun.includes('executions')) {
+      const executionsMigrationResult = await migrateExecutions(env, knex);
+      migrationSummary.executions = {
+        total_dynamo_db_records: executionsMigrationResult.total_dynamo_db_records,
+        migrated: executionsMigrationResult.migrated,
+        skipped: executionsMigrationResult.skipped,
+        failed: executionsMigrationResult.failed,
+      };
+    }
+
+    if (migrationsToRun.includes('granules')) {
+      const { granulesResult, filesResult } = await migrateGranulesAndFiles(
```

Comment:
If we're eventually going to be writing these outputs to S3, should we be recording the filter options in the output somehow?   Else if we run a bunch of these reports it might be hard to tell which collection you just tried to migrate.

Reply:
Not a bad idea, but I assume I would need for #2179 to merge first before I could add in that information?

Reply:
Not necessarily, if we updated the output to have a collection key, or something like that.       If we're planning on including it in the *filename* then I guess that should probably be coordinated with @botanical depending on merge order.      

Given you can't have a *list* of collections it might not make sense to have it in the report object itself, but I wouldn't hate it either given it would allow programmatic consumption of some range of report files.        Embedding it in the S3 key probably also suffices to address the concern though, insofar as it is a concern.

Reply:
Now that I look more closely, #2179 is really just producing a report of errors. It's not storing a general migration summary to S3.

So I can definitely add this information to the migration result information, but for now that's just going to show up in the CW log unless we do some further work. Thoughts?

Reply:
I made some initial changes: 4be478fc232b9bb9de014b6d5777dbf78ae86de1

But as I said, for now these would only show up in CW

Reply:
Right, re: 2179, that's just errors.    *frown* 

Given that, I think adding it to the object provides needed transparency on what the report output is referring to and we can punt actual outputs to a file as out-of-scope/YAGNI here.